### PR TITLE
Global cli version

### LINKED ISSUE #41: Convert the project into a CLI version
So we would install it globally and use on existing nextjs files.

Functionality:
`a0 gen 'description here'`
`npm i -g aleph0`

Options for v1:
- `rd` - regenerate description of prompt
- `-p` project directory (defaults to current working directory`
- ~~`-o` open ai key @elie222 WDYT?~~ use an ENV var for the `OPENAI__API_KEY`.

Future options:
- `-s` snippet dir
- `-m` model preference
- `-p` pricing estimation.
- `--debug` debug the output
- `-i` interactive decision before acitons.

### PR DESCRIPTION
**Pull request by agamm**
Fixes: https://github.com/alephmatic/nextjs-ai/issues/41

### CODE REVIEW

**Comment by agamm** [comment]
@elie222 How do you install it globally? (bun/pnpm?)
I'm getting weird errors that I can only fix with manual intervention.

File: aleph0/.github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        uses: actions/checkout@v2
+      - name: node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+          registry-url: https://registry.npmjs.org
+      - name: publish
+        run: npm publish --access public
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NPM_AUTH_TOKEN}}
```
Comment (elie222) [inline aleph0/.github/workflows/publish.yml]: doens't handle npm versions. think this fails unless package.json has updated version in it.

File: aleph0/README.md [hunk 2]
```diff
@@ -24,3 +24,8 @@ CONSOLA_LEVEL=4 bun src/index.ts gen "create a prisma schema file with a blog mo
 ```bash
 CONSOLA_LEVEL=4 bun run src/index.ts gen "add a file named agam.ts in the app/ folder" -p ../examples/next -rd false
 ```
+
+## Building
+
+`npm run build`
+`npm i -g .`
```
Comment (elie222) [inline aleph0/README.md]:
```suggestion
```bash
npm run build
npm i -g .
```
```

**Review by elie222 — DISMISSED** [review]

File: aleph0/.github/workflows/publish.yml (hunk 1 above)
Comment (agamm) [reply aleph0/.github/workflows/publish.yml]: I wanted to remove anyways :+1:

**Comment by agamm** [comment]
@elie222 WDYT?

**Comment by elie222** [comment]
> @elie222 WDYT?

sorry, missed the qu before. not working for me either. trying to play with it to fix it and run it locally.

**Comment by elie222** [comment]
@agamm my latest push allows me to build and then run with `node ./dist/index.js`. lodash/kebabCase import was causing problems.

`npx .` doesn't work yet though. and couldn't get it working with `pnpm dlx` either.

i need to look at it more tomorrow.

**Comment by agamm** [comment]
> @agamm my latest push allows me to build and then run with `node ./dist/index.js`. lodash/kebabCase import was causing problems.
> 
> `npx .` doesn't work yet though. and couldn't get it working with `pnpm dlx` either.
> 
> i need to look at it more tomorrow.

PXL_20231125_030552319

Will see tomorrow 🤠

**Pull request merged**